CanvasPlusPlayground/Common/Network/API Requests/GetPagesRequest.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  GetPagesRequest.swift
+//  CanvasPlusPlayground
+//
+//  Created by Max Ko on 2/21/25.
+//
+
+import Foundation
+
+struct GetPagesRequest: CacheableArrayAPIRequest {
+    typealias Subject = PageAPI
+
+    let courseId: String
+
+    var path: String { "courses/\(courseId)/pages" }
+
+    var queryParameters: [QueryParameter] {
+        [
+            ("sort", sort),
+            ("order", order),
+            ("search_term", searchTerm),
+            ("published", published),
+            ("per_page", perPage)
+        ]
+        + include.map { ("include[]", $0.rawValue) }
+    }
+
+    // MARK: Query Params
+    let sort: String?   // Allowed values: title, created_at, updated_at
+    let order: String?  // Allowed values: asc, desc
+    let searchTerm: String?
+    let published: Bool?
+    let include: [Include]
+    let perPage: Int
+
+    init(
+        courseId: String,
+        sort: String? = nil,
+        order: String? = nil,
+        searchTerm: String? = nil,
+        published: Bool? = nil,
+        include: [Include] = [],
+        perPage: Int = 50
+    ) {
+        self.courseId = courseId
+        self.sort = sort
+        self.order = order
+        self.searchTerm = searchTerm
+        self.published = published
+        self.include = include
+        self.perPage = perPage
+    }
+
+    var requestId: String? { courseId }
+    var requestIdKey: ParentKeyPath<Page, String?> { .createWritable(\.courseID) }
+    var idPredicate: Predicate<Page> {
+        #Predicate<Page> { page in
+            page.courseID == requestId
+        }
+    }
+
+    var customPredicate: Predicate<Page> {
+        let publishedPredicate: Predicate<Page>
+        if let published {
+            publishedPredicate = #Predicate<Page> { page in
+                page.published == published
+            }
+        } else {
+            publishedPredicate = .true
+        }
+
+        let searchTermPredicate: Predicate<Page>
+        if let searchTerm {
```

Comment:
I've seen issues in the past if this not nil but empty, so this is safer:
```suggestion
        if let searchTerm, !searchTerm.isEmpty {
```